### ISSUE TITLE
If I train the model_forward, will the weights in the model_all changed too?

### ISSUE THREAD

**Issue by JarmoHu** [keras-team/keras#13445]
- [✓] Check that you are up-to-date with the master branch of Keras. You can update with:

pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [✓] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [×] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [✓] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

''' python

    import scipy.io as sio
    import numpy as np
    from sklearn.model_selection import train_test_split

    import tensorflow as tf
    import math

    def load_data(data_address):
        data = sio.loadmat(data_address)

        pattern = data['allvc']
        predict = data['allt1']
        pattern = pattern.astype('float32')
        predict = predict.reshape(-1)
        pattern = pattern.reshape(32768, 25)
        predict = np.transpose(predict)
        x_train, x_test, y_train, y_test = train_test_split(pattern,
                                                            predict,
                                                            test_size=0.9,
                                                            random_state=0)
        return x_train, x_test, y_train, y_test

    def create_model():
        b1 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='b1',
                                   input_shape=(25, ))
        b2 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='b2',
                                   input_shape=(256, ))
        b3 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='b3',
                                   input_shape=(256, ))
        b4 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='b4',
                                   input_shape=(256, ))
        b5 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='b5',
                                   input_shape=(256, ))
        b6 = tf.keras.layers.Dense(25,
                                   activation='sigmoid',
                                   name='b6',
                                   input_shape=(256, ))
        f1 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='f1',
                                   input_shape=(25, ))
        f2 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='f2',
                                   input_shape=(256, ))
        f3 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='f3',
                                   input_shape=(256, ))
        f4 = tf.keras.layers.Dense(256,
                                   activation='sigmoid',
                                   name='f4',
                                   input_shape=(256, ))
        f5 = tf.keras.layers.Dense(1,
                                   activation='sigmoid',
                                   name='f5',
                                   input_shape=(256, ))

        model_all = tf.keras.Sequential()
        model_all.add(b1)
        model_all.add(b2)
        model_all.add(b3)
        model_all.add(b4)
        model_all.add(b5)
        model_all.add(b6)
        model_all.add(f1)
        model_all.add(f2)
        model_all.add(f3)
        model_all.add(f4)
        model_all.add(f5)

        model_forward = tf.keras.Sequential()
        model_forward.add(f1)
        model_forward.add(f2)
        model_forward.add(f3)
        model_forward.add(f4)
        model_forward.add(f5)

        return model_all, model_forward

'''
I want to train model_forward first, and then freeze the weights in it. Finally I want to train model_all with the layer named 'f1'-'f5' frozen. Is the code OK?

''' python

    model_all, model_forward = create_model()
    model_forward.compile(...)
    model_forward.fit(...)
    f1.trainable = False
    f2.trainable = False
    f3.trainable = False
    f4.trainable = False
    f5.trainable = False
    model_all.compile(...)
    model_all.fit(...)

'''
